node/pod.go

```diff
@@ -216,9 +216,26 @@ func (pc *PodController) updatePodStatus(ctx context.Context, podFromKubernetes
 	kPod.Lock()
 	podFromProvider := kPod.lastPodStatusReceivedFromProvider.DeepCopy()
 	kPod.Unlock()
-	if cmp.Equal(podFromKubernetes.Status, podFromProvider.Status) {
+	if cmp.Equal(podFromKubernetes.Status, podFromProvider.Status) && podFromProvider.DeletionTimestamp == nil {
 		return nil
 	}
+	if podFromProvider.DeletionTimestamp != nil && podFromKubernetes.DeletionTimestamp == nil {
+		deleteOptions := metav1.DeleteOptions{
+			GracePeriodSeconds: podFromProvider.DeletionGracePeriodSeconds,
+		}
+		if podFromProvider.DeletionTimestamp.After(time.Now()) {
+			deleteOptions.GracePeriodSeconds = new(int64)
+		}
+		// check status here to avoid pod re-recreated deleting incorrectly
+		if cmp.Equal(podFromKubernetes.Status, podFromProvider.Status) {
```

Comment:
Why this comparison?

Reply:
If we delete a pod from K8s，re-create a pod with same name. We may delete the pod re-created in K8s incorrectly